### DISCUSSION TITLE
Third party use of plugins and plugin conflicts in v9

### DISCUSSION THREAD

**Discussion by dpickett** [eslint/eslint#19056] [Q&A]
We maintain a set of linting rules that we use across all of our projects - https://github.com/LaunchWare/eslint-config-node and https://github.com/LaunchWare/eslint-config-react

We're also big fans of the NX system for managing monorepos. NX maintains a set of rules to enforce good monorepo practices. (https://github.com/nrwl/nx/tree/master/packages/eslint-plugin)

Both npm's declare a plugin property for `typescript-eslint`. When I try to integrate both libraries into our consuming applications, I run into a blocking problem. 

`Cannot redefine plugin "@typescript-eslint"`. I could remove the plugin directive from our shared config libraries, but then I can't refine linting rules in the consuming application or in our shared config. I'd like for both libraries to refine linting rules for the plugin, but I'm not sure how to do that in v9.

I'm thinking this will be a problem for many users of eslint. How does the team recommend handling these plugin dependencies and rule refinements moving forward?

**Comment by nzakas** [comment]
This means you are using two different versions of `typescript-eslint`. Basically, you're ending up with something like this:

```js
[{
    plugins: { "typescript-eslint": tsESLint1,
},
{
    plugins: { "typescript-eslint": tsESLint2,
}]
```

So, first check to make sure you're using the same version of `typescript-eslint` as the Nx plugin.

Another option is to export more than one config: one that contains a definition for `typescript-eslint` and one that doesn't.

**Reply by dpickett** [reply]
I verified I was using a single version with `pnpm why typescript-eslint` and only one version showed up. If that's intended behavior should I file a bug in the eslint repository? I can try to build a reproducible repository.

**Reply by nzakas** [reply]
The only other thing I can think of is some config you're including might be using the [compat utilities](https://eslint.org/blog/2024/05/eslint-compatibility-utilities/) to wrap the plugin, in which case you'd need to wrap it as well. 

If you can come up with a repro case we can look at, go ahead and open an issue.
